Announce opam 2.2.1 (#104)
ocaml/platform-blog: opam-2-2-1.md

title: "opam 2.2.1 release"
authors: [
  "Raja Boujbel - OCamlPro" {"mailto:raja.boujbel(à)ocamlpro.com"}
  "Kate Deplaix - Ahrefs" {"mailto:kit-ty-kate(à)outlook.com"}
  "David Allsopp - Tarides" {"mailto:david(à)tarides.com"}
]
date: "2024-08-21"
--BODY--

_Feedback on this post is welcomed on [Discuss](https://discuss.ocaml.org/t/ann-opam-2-2-1-release/XXXXX)!_

We are pleased to announce the release of opam 2.2.1.

We've fixed a couple of regressions and would like to encourage users of opam 2.2 to upgrade.

## Changes

The three main changes are the following:

* Fix a regression in `opam install --deps-only` where the direct dependencies were not set as root packages
* Fix a regression when building git packages where the build result were sometimes inconsistent compared to outside opam,

>   by disabling shallow clone by default except when fetching an opam repositories

Reviewer: ```suggestion
  by disabling shallow clone by default except when fetching an opam repositories [[#6145](https://github.com/ocaml/opam/issues/6145)]
```

Author: issue numbers were never shown in previous blog posts. I don't think there is much value in linking them in there

Reviewer: They were shown for minor releases, cf. 2.1.x, 2.0.x, ...

Author: it made sense before because of our backporting protocol (on PR for everything) but now i don't think it makes sense

Reviewer: It's independent from the backporting protocol. It was added in the first place to highlight the issue that is fixed, as minor releases contain (usually) only fixes -without having to go to a more complete change/release note.

Author: fair enough. Done. I've also added a highlight for the `--deps-only` regression even though it's not the issue that triggered the fix

* Mitigate [curl/curl#13845](https://github.com/curl/curl/issues/13845) by falling back from `--write-out` to `--fail`

>   if exit code 43 is returned by curl

Reviewer: ```suggestion
  if exit code 43 is returned by curl [[#6120](https://github.com/ocaml/opam/issues/6120)]
```

title: "opam 2.2.1 release"
authors: [
  "Raja Boujbel - OCamlPro" {"mailto:raja.boujbel(à)ocamlpro.com"}
  "Kate Deplaix - Ahrefs" {"mailto:kit-ty-kate(à)outlook.com"}
  "David Allsopp - Tarides" {"mailto:david(à)tarides.com"}
]

> date: "2024-08-21"

Reviewer: ```suggestion
date: "2024-08-22"
```

title: "opam 2.2.1 release"
authors: [
  "Raja Boujbel - OCamlPro" {"mailto:raja.boujbel(à)ocamlpro.com"}
  "Kate Deplaix - Ahrefs" {"mailto:kit-ty-kate(à)outlook.com"}
  "David Allsopp - Tarides" {"mailto:david(à)tarides.com"}
]
date: "2024-08-21"
--BODY--

_Feedback on this post is welcomed on [Discuss](https://discuss.ocaml.org/t/ann-opam-2-2-1-release/XXXXX)!_

We are pleased to announce the release of opam 2.2.1.

We've fixed a couple of regressions and would like to encourage users of opam 2.2 to upgrade.

## Changes

The three main changes are the following:

* Fix a regression in `opam install --deps-only` where the direct dependencies were not set as root packages

>   ([discuss.ocaml.org#15142](https://discuss.ocaml.org/t/how-to-list-all-root-dependencies-in-the-current-switch/15142))

Reviewer 2: ```suggestion
  ([reported on Discuss](https://discuss.ocaml.org/t/how-to-list-all-root-dependencies-in-the-current-switch/15142))
```
?

Reviewer 2: @rjbou - is there an issue or the like we can reference in addition to the Discuss thread for what led you to spot the regression?

* Fix a regression when building git packages where the build result were sometimes inconsistent compared to outside opam,

>   by disabling shallow clone by default except when fetching an opam repositories

Reviewer 3: Awesome! I'm amazed by your quick response from the issue report to a fix and now even a release! Thanks so much for your work on opam!

> * Fix a regression when building git packages where the build result were sometimes inconsistent compared to outside opam,
>   by disabling shallow clone by default except when fetching an opam repositories

Reviewer: ```suggestion
* Fix a regression when fetching git packages where the resulting git repository could lead to unexpected outputs of git commands, by disabling shallow clone by default except when fetching an opam repositories
```

  ([#6145](https://github.com/ocaml/opam/issues/6145))
* Mitigate [curl/curl#13845](https://github.com/curl/curl/issues/13845) by falling back from `--write-out` to `--fail`

>   if exit code 43 is returned by curl

Reviewer 2: ```suggestion
  if exit code 43 is returned by curl. In particular, this fixes `opam init` when run from cmd/PowerShell on Windows 11 23H2
```

  ([#6120](https://github.com/ocaml/opam/issues/6120))

A couple more improvements and additions to the testsuite were made.
You can view the full list of changes in the [release note](https://github.com/ocaml/opam/releases/tag/2.2.1).

## Try it!

The upgrade instructions are unchanged:

1. Either from binaries: run

For Unix systems
```
bash -c "sh <(curl -fsSL https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh) --version 2.2.1"
```
or from PowerShell for Windows systems
```
Invoke-Expression "& { $(Invoke-RestMethod https://raw.githubusercontent.com/ocaml/opam/master/shell/install.ps1) }"
```

> or download manually from [the Github "Releases" page](https://github.com/ocaml/opam/releases/tag/2.2.1) to your PATH.

Reviewer 2: ```suggestion
(or via `winget upgrade OCaml.opam`) or download manually from [the Github "Releases" page](https://github.com/ocaml/opam/releases/tag/2.2.1) to your PATH.
```
We can get the PR in for winget as soon as the signed binary is up. Seems worth listing this approach given that winget literally downloads the binary from our release tags?

Reviewer: If it can be published before publishing the blog post, sure!

Reviewer: It is published!
